src/libraries/System.IO.Compression/tests/CompressionStreamUnitTests.Gzip.cs

```diff
@@ -274,6 +274,51 @@ public void DerivedStream_ReadWriteSpan_UsesReadWriteArray()
             }
         }
 
+        [Fact]
+        public void StreamCorruption_IsDetected()
+        {
+            byte[] source = Enumerable.Range(0, 64).Select(i => (byte)i).ToArray();
+            var buffer = new byte[64];
+            byte[] compressedData;
+            using (var compressed = new MemoryStream())
+            using (Stream compressor = CreateStream(compressed, CompressionMode.Compress))
+            {
+                foreach (byte b in source)
+                {
+                    compressor.WriteByte(b);
+                }
+
+                compressor.Dispose();
+                compressedData = compressed.ToArray();
+            }
+
+            // corrupting these bytes goes undetected by gzip, skip them
+            int[] byteToSkip = { 3, 4, 5, 6, 7, 8, 9 };
```

Comment:
Can you please add, as a comment, the reason why these bytes don't corrupt the compressed data?

Reply:
Ok, done.